client/components/update-post-status/update-template.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+export default function UpdateTemplate( {
+	post,
+	previousStatus,
+	resetToPreviousState,
+	status,
+	strings: s,
+} ) {
+	let updateClass = 'conf-alert';
+	let updateText, undoTemplate, undoClick, trashText;
+
+	switch ( status ) {
+		case 'trashing':
+		case 'deleting':
+			trashText = ( status === 'deleting' ) ? s.deleting : s.trashing;
+			updateText = (
+				<span>{ trashText } <span className="loading-dot">.</span><span className="loading-dot">.</span><span className="loading-dot">.</span></span>
+			);
+			updateClass += ' conf-alert--trashing';
+			break;
+
+		case 'trash':
+			undoClick = resetToPreviousState;
+			undoTemplate = (
+				<a className="undo" onClick={ undoClick }><span>{ s.undo }</span></a>
+			);
+			updateText = s.trashed;
```

Comment:
Yeah. Post types and i18n are hard — or easy but very inflexible. There's a lot of reasons why that wouldn't go well. Example off the top of my head:

> Delete this %(postType)s permanently?

→ Apagar **este** artigo para sempre?
→ Apagar **esta** página para sempre?

This would be `pt-pt`, which features gendered nouns, like many other Romance languages, but other features can break this as well. I think the current impl is a _good enough_ solution that I'd like to leave untouched for now.